### ISSUE TITLE
CRD Helm chart v0.37.0-v1.0.0 release issues

### ISSUE THREAD

**Issue by pvickery-ParamountCommerce** [aws/karpenter-provider-aws#6899]
### Description

**Observed Behavior**:
[Conversion webhooks are added to the CRDs in v0.37.0](https://github.com/aws/karpenter-provider-aws/blob/5bdf9c3a9d6b01dc6934c3807dfa69160d411131/charts/karpenter-crd/values.yaml#L3) but [they were not supposed to be added until v1.0.0
](https://karpenter.sh/docs/upgrading/upgrade-guide/#upgrading-to-100)

This looks related to https://github.com/aws/karpenter-provider-aws/issues/6897 and https://github.com/aws/karpenter-provider-aws/issues/6889

**Expected Behavior**:
No conversion webhooks are used in karpenter-crd v0.37.0

**Reproduction Steps** (Please include YAML):
Install karpenter-crd v0.37.0 and karpenter v0.37.0. Errors will occur as follows since karpenter v0.37.0 does not use port 8443

> failed to load initial state of resource EC2NodeClass.karpenter.k8s.aws: conversion webhook for karpenter.k8s.aws/v1beta1, Kind=EC2NodeClass failed: Post "https://karpenter.kube-system.svc:8443/conversion/karpenter.k8s.aws?timeout=30s": no service port 8443 found for service "karpenter"

**Versions**:
- Chart Version: v0.37.0
- Kubernetes Version (`kubectl version`): any

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by nlpfollower** [comment]
I'm having a similar issue running the Getting Started guide (https://karpenter.sh/preview/getting-started/getting-started-with-karpenter/)

running the command `kubectl scale deployment inflate --replicas 5` results in the error below:

```
{"level":"INFO","time":"2024-08-30T18:33:56.064Z","logger":"controller","message":"Starting workers","commit":"5bdf9c3","controller":"nodeclaim.lifecycle","controllerGroup":"karpenter.sh","controllerKind":"NodeClaim","worker count":1000}
{"level":"INFO","time":"2024-08-30T18:33:56.065Z","logger":"controller","message":"Starting workers","commit":"5bdf9c3","controller":"nodeclaim.consistency","controllerGroup":"karpenter.sh","controllerKind":"NodeClaim","worker count":10}
{"level":"INFO","time":"2024-08-30T18:33:56.065Z","logger":"controller","message":"Starting workers","commit":"5bdf9c3","controller":"nodeclaim.termination","controllerGroup":"karpenter.sh","controllerKind":"NodeClaim","worker count":100}
{"level":"INFO","time":"2024-08-30T18:33:56.065Z","logger":"controller","message":"Starting workers","commit":"5bdf9c3","controller":"status","controllerGroup":"karpenter.sh","controllerKind":"NodePool","worker count":10}
{"level":"INFO","time":"2024-08-30T18:33:56.065Z","logger":"controller","message":"Starting workers","commit":"5bdf9c3","controller":"nodeclaim.disruption","controllerGroup":"karpenter.sh","controllerKind":"NodeClaim","worker count":10}
{"level":"INFO","time":"2024-08-30T18:33:56.066Z","logger":"controller","message":"Starting workers","commit":"5bdf9c3","controller":"status","controllerGroup":"karpenter.sh","controllerKind":"NodeClaim","worker count":10}
{"level":"ERROR","time":"2024-08-30T18:33:56.385Z","logger":"webhook.ConversionWebhook","message":"Reconcile error","commit":"5bdf9c3","knative.dev/traceid":"f6d15b03-34d8-4b4f-acf1-7e1211dbb9e1","knative.dev/key":"ec2nodeclasses.karpenter.k8s.aws","duration":"454.85068ms","error":"failed to update webhook: Operation cannot be fulfilled on customresourcedefinitions.apiextensions.k8s.io \"ec2nodeclasses.karpenter.k8s.aws\": the object has been modified; please apply your changes to the latest version and try again"}
```

**Comment by booleanbetrayal** [comment]
@pvickery-ParamountCommerce - you may wish to follow the thread @ #6847

**Comment by engedaam** [comment]
The conversion webhook block was added to support rollback from v1.0. You are able to disable the webhooks by using the values of `webhook.enabled=false`

**Comment by pvickery-ParamountCommerce** [comment]
> The conversion webhook block was added to support rollback from v1.0. You are able to disable the webhooks by using the values of `webhook.enabled=false`

The default values should not cause an upgrade to break so I think `webhook.enabled=false` should be the default. At the very least, this information should be included in [the notes for upgrading to 0.37.0+](url).

**Comment by engedaam** [comment]
With updated webhook defaults, in the patch version. This is mitigated:
   - v0.37.3
   - v0.36.5
   - v0.35.8
   - v0.34.9
   - v0.33.8
